apps/web/src/components/currency/index.tsx

```diff
@@ -22,8 +25,22 @@ export default function Currency({
   locale,
 }: ICurrency) {
   const globalLocale = useLocale()
-  const globalCurrency = useCurrency()
+  const { currency: globalCurrency } = useCurrency()
   const { conversionRate } = useI18n()
+  const [hydrated, setHydrated] = useState(false)
+
+  if (!hydrated) {
+    currency = DEFAULT_CURRENCY
+    locale = DEFAULT_LOCALE
+  }
+
+  useEffect(() => {
+    /*
+     * This addresses an issue where the user's selected currency is not
+     * available at hydration time, causing a discrepancy between server and client
+     */
+    setHydrated(true)
+  }, [])
```

Comment:
🎉

Reply:
Ended up removing this in favor of a fix in currency-context.